cmd/ctr/commands/snapshots/snapshots.go

```diff
@@ -67,6 +74,112 @@ var listCommand = cli.Command{
 	},
 }
 
+var diffCommand = cli.Command{
+	Name:      "diff",
+	Usage:     "get the diff of two snapshots. the default second snapshot is the first snapshot's parent.",
+	ArgsUsage: "[flags] <idA> [<idB>]",
+	Flags: append([]cli.Flag{
+		cli.StringFlag{
+			Name:  "media-type",
+			Usage: "media type to use for creating diff",
+		},
+		cli.StringFlag{
+			Name:  "ref",
+			Usage: "content upload reference to use",
+		},
+		cli.BoolFlag{
+			Name:  "keep",
+			Usage: "keep diff content. up to creator to delete it.",
+		},
+	}, commands.LabelFlag),
+	Action: func(context *cli.Context) error {
+		var (
+			idA = context.Args().First()
+			idB = context.Args().Get(1)
+		)
+		if idA == "" {
+			return errors.New("snapshot id must be provided")
+		}
+		client, ctx, cancel, err := commands.NewClient(context)
+		if err != nil {
+			return err
+		}
+		defer cancel()
+
+		ctx, done, err := client.WithLease(ctx)
+		if err != nil {
+			return err
+		}
+		defer done()
+
+		var desc ocispec.Descriptor
+		labels := commands.LabelArgs(context.StringSlice("label"))
+		snapshotter := client.SnapshotService(context.GlobalString("snapshotter"))
+
+		if context.Bool("keep") {
+			labels["containerd.io/gc.root"] = time.Now().UTC().Format(time.RFC3339)
```

Comment:
do we need this or will leases work here?

Reply:
I suggested this as a way to keep around the diffs created by the cli and leave it up to the creator to remove later. By default the lease will keep the diff around for the duration of the cli command (while it gets copied to stdout), but it will get cleaned up after it completes.